castle/src/main/java/io/castle/android/queue/EventQueue.java

```diff
@@ -58,17 +64,27 @@ private synchronized File getFile(Context context) {
     }
 
     private synchronized void init(Context context) throws IOException {
+        executor = Executors.newSingleThreadExecutor();
```

Comment:
I think passing executor from Castle::setup will be a more flexible and pretty chip to implement, just one another Castle::setup method with executor in parameters.
But maybe it's redundant :)

Reply:
The reasoning I had for doing it this way is to make sure that there is always just one queue operation running concurrently. Being able to set the executor on your own would make it possible to provide a multi thread executor and it feels unnecessary and hard to test for our use case.